src/fides/api/models/privacy_experience.py

```diff
@@ -97,36 +189,111 @@ def create(
+        record combining details from the translation and the Experience Config
+        - Links regions to the ExperienceConfig (by creating/updating associated PrivacyExperiences
+        and adding a FK back to the config)
+        - Adds/removes Notices to the Experience Config
+        """
+        translations = data.pop("translations", [])
+        regions = data.pop("regions", [])
+        privacy_notices = data.pop("privacy_notices", [])
+        data.pop(
+            "id", None
+        )  # Default templates may have ids but we don't want to use them here
+
         experience_config: PrivacyExperienceConfig = super().create(
             db=db, data=data, check_name=check_name
         )
-        # create the history after the initial object creation succeeds, to avoid
-        # writing history if the creation fails and so that we can get the generated ID
-        data.pop(
-            "id", None
-        )  # Default Experience Configs have id specified but we don't want to use the same id for the historical record
-        history_data = {
-            **data,
-            "experience_config_id": experience_config.id,
-        }
-        PrivacyExperienceConfigHistory.create(db, data=history_data, check_name=False)
+
+        for translation_data in translations:
+            data.pop(
+                "id", None
+            )  # Default Experience Configs have id specified but we don't want to use the same id for the historical record
+            translation = ExperienceTranslation.create(
+                db,
+                data={**translation_data, "experience_config_id": experience_config.id},
+            )
+            PrivacyExperienceConfigHistory.create(
+                db,
+                data={**data, **translation_data, "translation_id": translation.id},
+                check_name=False,
+            )
+
+        upsert_privacy_experiences_after_config_update(db, experience_config, regions)
+        link_notices_to_experience_config(
+            db, notice_keys=privacy_notices, experience_config=experience_config
+        )
+
         return experience_config
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperienceConfig:
         """
-        Overrides the base update method to automatically bump the version of the
-        PrivacyExperienceConfig record and also create a new PrivacyExperienceConfigHistory entry
+        Updates a PrivacyExperienceConfig which is a large exercise!
+
+        - Updates the PrivacyExperienceConfig details
+        - For each supplied translation, add, update, or delete translations so they match
+        the translations in the update request
+        - For each remaining translation, if the translation or the config has changed,
+        create a historical record for auditing purposes
+        - Link or unlink regions (via Privacy Experiences)
+        - Link or unlink Privacy Notices
```

Comment:
Tentative drafts of the PrivacyExperienceConfig update method.  These are monster methods that create/update/delete numerous resources across multiple tables.  Further thought needed into the best way to do this. The create method has similar concerns.

Reply:
less concerned about performance here, these methods are for setting up privacy experiences, something that will occur by the customer occasionally. 

It's that normally don't like one model to be updating a bunch of other models (maybe hooks are better here), and there's also a chance something happens midway through and only some of these models are updated. 

But the more I've thought about it, these are core updates that need to stay together in one place, so I'm leaning towards leaving as-is